FitGoal/FitGoal/ExerciseDetailViewController/ExercisePreviewCell.swift

```diff
@@ -0,0 +1,232 @@
+//
+//  ExercisePreviewCell.swift
+//  FitGoal
+//
+//  Created by Eliany Barrera on 13/3/20.
+//  Copyright © 2020 Eliany Barrera. All rights reserved.
+//
+
+import UIKit
+
+class ExercisePreviewCell: UICollectionViewCell {
+    
+    static var identifier = "Exercise Preview"
+    
+    private var imageLoadingState: ImageLoadingState = .inProgress {
+        didSet {
+            switch imageLoadingState {
+            case .inProgress:
+                imageGradient.isHidden = true
+                playButton.isHidden = true
+                placeholder.startAnimating()
+            case .finished(let image):
+                imageGradient.isHidden = false
+                playButton.isHidden = false
+                placeholder.stopAnimating()
+                exerciseImage.image = image
+            case .failed(let error):
+                print("Unable to load image with error: \(error)")
+            }
+        }
+    }
+    
+    var exercise: Exercise? {
+        didSet {
+            imageURL = exercise?.url
+            guard let title = exercise?.name else { return }
+            
+            cellTitle.attributedText = title.uppercased().formattedText(
+                font: "Oswald-Medium",
+                size: 34,
+                color: .white,
+                kern: -0.14
+            )
+        }
+    }
+    
+    private var currentImageDownloadTask: URLSessionTask?
+    
+    private var imageURL: URL? {
+        didSet {
+            guard let imageURL = imageURL else {
+                return
+            }
+            currentImageDownloadTask = imageURL.fetchImage { result in
```

Comment:
Why do you need the `currentImageDownloadTask` in this case? You are never canceling that task so I think you don't need it at all. After all, in this case the cell isn't reused so there is no need to cancel the request.

Reply:
I just copied and paste code, have to be more careful with this, though. I'm seeing it is not always what you need.